--- g-wrap-1.9.7/configure.ac.orig	2006-11-05 14:57:30.000000000 +0100
+++ g-wrap-1.9.7/configure.ac	2006-12-30 02:10:17.463942711 +0100
@@ -119,12 +119,7 @@
 AC_SUBST(GW_LANGUAGES)
 
 dnl Check for optional libraries
-AC_CHECK_PROG(have_glib, glib-config, yes, no)
-if test "$have_glib" = yes; then
-  PKG_CHECK_MODULES(GLIB, glib, [echo building GLib 1.2 support])
-else
-  echo Not building Glib 1.2 support
-fi
+PKG_CHECK_MODULES(GLIB, glib-2.0, [have_glib=yes; echo building GLib 2.0 support], [have_glib=no])
 AM_CONDITIONAL(HAVE_GLIB, test "$have_glib" = yes)
 
 
