diff -ur g-wrap-1.9.6-o/configure.ac g-wrap-1.9.6/configure.ac
--- g-wrap-1.9.6-o/configure.ac	2005-04-14 08:57:01.000000000 -0600
+++ g-wrap-1.9.6/configure.ac	2006-05-19 13:43:31.000000000 -0600
@@ -95,7 +95,7 @@
 AM_CONDITIONAL(HAVE_SRFI_35, test "$HAVE_SRFI_35" = yes)
 
 dnl Check for optional libraries
-PKG_CHECK_MODULES(GLIB, glib, [have_glib=yes], [have_glib=no])
+PKG_CHECK_MODULES(GLIB, glib2, [have_glib=yes], [have_glib=no])
 AM_CONDITIONAL(HAVE_GLIB, test "$have_glib" = yes)
 
 dnl Check for compiler flags and warnings. This should be last, since
