--- g-wrap-1.9.11/guile/g-wrap-2.0-guile.pc.in.orig	2007-08-14 15:21:29.000000000 +0200
+++ g-wrap-1.9.11/guile/g-wrap-2.0-guile.pc.in	2008-05-28 23:43:28.632623606 +0200
@@ -11,5 +11,6 @@
 Name: G-Wrap
 Description: A Guile wrapper generator
 Version: @VERSION@
-Libs: -L${libdir} -lgwrap-guile-runtime -lgwrap-core-runtime @GUILE_LDFLAGS@ @LIBFFI_LIBS_INSTALLED@
-Cflags: @STD_CFLAGS@ -I${includedir} @GUILE_CFLAGS@ @LIBFFI_CFLAGS_INSTALLED@
+Requires: libffi
+Libs: -L${libdir} -lgwrap-guile-runtime -lgwrap-core-runtime @GUILE_LDFLAGS@
+Cflags: @STD_CFLAGS@ -I${includedir} @GUILE_CFLAGS@
